### DISCUSSION TITLE
Snippets with links to other markdown files - other markdown files not rendering

### DISCUSSION THREAD

**Discussion by jasonnovichRunAI** [facelessuser/pymdown-extensions#1985] [Q&A] [answered]
Hi,
I got the snippets to work, and for the most part it does what I need it to do.
However, my snippet file is a shared file with 4 other files, basically content reuse.
In my snippet file I have links to other markdown files that are in the repo.
When I call the snippet file, the links to the markdown files appear as links, but the path to the file is not correct.

For example in my snippet file I have the following:
[Link to file](../docs/files/linked_file.md)

I call the snippet in a file that is located in another directory, for example:
docs/site_files/file_1.md (calles the snippet file above).

When the snippet is called the [Link to file] comes out blue like a link, but the URL for the file is:
/files/linked_file/sitefiles/file_1.md
Which does not show any file (in my case leads to one of my default error pages of "file not found".

Is there any way to include MD files with a link in the snippet file?

Thanks!

**Comment by facelessuser** [comment, marked as answer]
I need a real example. You've just provided a link to my snippet which I use in my documents. Please create a simple zip that reproduces the issue.

**Reply by jasonnovichRunAI** [reply]
OK, I have recreated the issue. I could not get the nav to work, but that is not my issue, so please ignore. To see the issue after you build, go to page http://<build address>/folder1/file1/. You should see the "This is text from a linked file". If you click the link to the file you will get a 404 error, and the resulting link is http://<build address>/folder1/docs/folder2/file2.md. What I would like to happen is have file2 rendered as HTML.
[mkdocs snippet failing with links.zip](https://github.com/facelessuser/pymdown-extensions/files/10979110/mkdocs.snippet.failing.with.links.zip)

Thanks! 8-)

**Reply by facelessuser** [reply]
Snippets does fairly simple dumb replacements. Keep in mind that it is a Python Markdown plugin, it has no knowledge of MkDocs or relations of MkDocs directory structures. Also, it doesn't do any process normal Markdown before it does the replacements, so it doesn't touch links and has no knowledge of them as it is a preprocessor that runs before most other Markdown processes.

So, when you specify a link in a snippet, you need to specify the link relative to the file it is being inserted in. As mentioned before, it doesn't know or care about anything up at the MkDocs level. If you need a MkDocs navigational aware snippets handler, you would need a MkDocs plugin, not a Python Markdown plugin.

In short, replacing the link with `This is text from [a linked file](../folder2/file2.md)` got it to work fine.

**Reply by jasonnovichRunAI** [reply]
Great!! All working as expected now! Thank you for helping out!!
